packages/cli/src/commands/deploy/index.ts

```diff
@@ -174,6 +175,13 @@ export default async (client: Client): Promise<number> => {
   let { path: cwd } = pathValidation;
   const autoConfirm = argv['--yes'];
 
+  try {
+    await validateNpmrc(cwd);
+  } catch (err) {
+    prettyError(err);
+    return 1;
+  }
```

Comment:
This validation should happen in `vc build` (not `vc deploy`).

We might even want to move it all the way to `@vercel/build-utils` which where we invoke `npm install`

Reply:
Happy to add this validation to `vc build`, but I think we should keep it in `vc deploy` just to fail fast.

Reply:
Agreed. OK, then I can move the validate function to `build-utils`, then call it wherever we install dependencies. This won't fail fast, but it should cover all the bases and improve the error message.